pkg/apis/core/validation/validation.go

```diff
@@ -4534,7 +4534,10 @@ func validateServiceInternalTrafficFieldsValue(service *core.Service) field.Erro
 
 	if utilfeature.DefaultFeatureGate.Enabled(features.ServiceInternalTrafficPolicy) {
 		if service.Spec.InternalTrafficPolicy == nil {
-			allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("internalTrafficPolicy"), ""))
+			if service.Spec.Type == core.ServiceTypeNodePort ||
+				service.Spec.Type == core.ServiceTypeLoadBalancer || service.Spec.Type == core.ServiceTypeClusterIP {
+				allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("internalTrafficPolicy"), ""))
+			}
```

Comment:
IF we do this, we should have a comment here about why we do not enforce the opposite.  E.g. 

// We do not forbid ITP on other types because of historical reasons.  We did not check that before it went beta and we don't want to invalidate stored objects.

Reply:
Good call, let me add that comment